tests/scripts/all.sh

```diff
@@ -897,26 +897,49 @@ component_test_full_cmake_clang () {
     if_build_succeeded env OPENSSL_CMD="$OPENSSL_NEXT" tests/compat.sh -e '^$' -f 'ARIA\|CHACHA'
 }
 
-component_build_deprecated () {
-    msg "build: make, full config + DEPRECATED_WARNING, gcc -O" # ~ 30s
+component_test_default_no_deprecated () {
+    msg "build: make, default + MBEDTLS_DEPRECATED_REMOVED" # ~ 30s
+    scripts/config.py set MBEDTLS_DEPRECATED_REMOVED
+    make CC=gcc CFLAGS='-O -Werror -Wall -Wextra'
+
+    msg "test: make, default + MBEDTLS_DEPRECATED_REMOVED" # ~ 5s
+    make test
+}
+
+component_test_full_no_deprecated () {
+    msg "build: make, full_non_deprecated config" # ~ 30s
+    scripts/config.py full_non_deprecated
+    make CC=gcc CFLAGS='-O -Werror -Wall -Wextra'
+
+    msg "test: make, full_non_deprecated config" # ~ 5s
+    make test
+}
+
+component_test_no_deprecated_warning () {
+    msg "build: make, full_non_deprecated config, MBEDTLS_DEPRECATED_WARNING" # ~ 30s
+    scripts/config.py full_non_deprecated
+    scripts/config.py unset MBEDTLS_DEPRECATED_REMOVED
```

Comment:
The goal is to test the effect of `MBEDTLS_DEPRECATED_WARNING`. If `MBEDTLS_DEPRECATED_REMOVED` is also set, `MBEDTLS_DEPRECATED_WARNING` doesn't have any effect because the corresponding code is not compiled at all.

Reply:
Deprecated functions should be guarded by `#if !defined(MBEDTLS_DEPRECATED_REMOVED)`, and have a `__attribute__((deprecated))` guarded by `#if defined(MBEDTLS_DEPRECATED_WARNING)`. Testing with both `MBEDTLS_DEPRECATED_REMOVED` and `MBEDTLS_DEPRECATED_WARNING` wouldn't be very useful since all the difference that `WARNING` makes is supposed to be guarded by `REMOVED`. Testing with `WARNING` but without `REMOVED` checks that there isn't a deprecated feature that isn't guarded by `REMOVED`.

Reply:
In that case isn't possible to start from "full" (rather than (full without deprecated stuff) + deprecated stuff) land to set MBEDTLS_DEPRECATED_WARNING?

Reply:
That wouldn't result in the same configuration. Which I think indicates that we don't quite do things right when it comes to disabling deprecated things: we do it differently for feature options and for individual functions. If you turn on `DEPRECATED_REMOVED`, this skips definitions of deprecated functions, but it causes an error if you enable a deprecated feature option.

I don't think I want to tackle this can of worms in this PR which was mainly about `full` itself, and only changed the `deprecated` component of `all.sh` because I had to due to some deprecated things now being in `full`. And I'm not even sure I want to tackle it before Mbed TLS 3. Or until we have this better configuration interface that we've been talking about for years, whichever comes first.

Reply:
ok thanks for the explanation.